cpp/src/Glacier2/RoutingTable.cpp

```diff
@@ -125,7 +125,7 @@ Glacier2::RoutingTable::add(const ObjectProxySeq& unfiltered, const Current& cur
     return evictedProxies;
 }
 
-ObjectPrxPtr
+optional<ObjectPrx>
 Glacier2::RoutingTable::get(const Identity& ident)
 {
     if(ident.name.empty())
```

Comment:
Since it's an internal function, this should never happen. We must catch and reject such invalid identity before reaching this internal function. See #1813.